sasmodels/sasview_model.py

```diff
@@ -56,15 +56,16 @@ def _register_old_models():
 
     import sas.models
     from sasmodels.conversion_table import CONVERSION_TABLE
-    for new_name, conversion in CONVERSION_TABLE.items():
-        # CoreShellEllipsoidModel => core_shell_ellipsoid:1
-        new_name = new_name.split(':')[0]
-        old_name = conversion[0]
-        module_attrs = {old_name: find_model(new_name)}
-        ConstructedModule = type(old_name, (), module_attrs)
-        old_path = 'sas.models.' + old_name
-        setattr(sas.models, old_path, ConstructedModule)
-        sys.modules[old_path] = ConstructedModule
+    for version, _ in sorted(CONVERSION_TABLE.iteritems()):
+        for new_name, conversion in CONVERSION_TABLE.get(version).items():
+            # CoreShellEllipsoidModel => core_shell_ellipsoid:1
+            new_name = new_name.split(':')[0]
+            old_name = conversion[0]
```

Comment:
The names in the conversion table now refer to the name stored in the page state file (presumably) rather than the name of the model class in sasview 3.1.x.  So for example, peak_lorentz has name "Peak Lorentz Model".

The point of this code is to provide a namespace of old sasview model names so that sum/product models created in 3.x can be used in 4.x, which requires class names rather than model names.

Loading a saved analysis which from 3.x which uses a sum/product model presents additional challenges, since the new models will have parameters with the new names rather than the old.  You would need to poke around in the old model definition file from the plugins directory, find which models were being summed, and translate the parameters for them.  Make it recursive so that sums of sums are loadable.  This will not be achievable with reasonable effort.